crates/ruma-federation-api/src/authenticated_media.rs

```diff
@@ -172,12 +172,20 @@ fn try_from_multipart_mixed_response<T: AsRef<[u8]>>(
     let mut full_boundary = Vec::with_capacity(boundary.len() + 4);
     full_boundary.extend_from_slice(b"\r\n--");
     full_boundary.extend_from_slice(boundary);
+    let full_boundary_no_crlf = &full_boundary[2..];
```

Comment:
Indexing into data like that always makes me uneasy, how about:

```suggestion
    let full_boundary_no_crlf = full_boundary.strip_prefix(b"\r\n").unwrap();
```